### ISSUE TITLE
Add support for Cookie authorization

### ISSUE THREAD

**Issue by ereOn** [wunderfrucht/gouqi#54]
Right now `gouqi` support basic and bearer authentication methods, but Jira actually also supports Cookie-based authentication.

<img width="356" alt="image" src="https://github.com/wunderfrucht/gouqi/assets/585131/04673cb9-d9c7-4178-a61f-9bfd1af48b94">

Our organization sadly only allows cookie-based auth. even in an automated context... Would you be willing to accept a PR that adds support for this? I could tackle this fairly quickly if you agree that this would be a nice addition.

**Comment by avrabe** [comment]
Yes, would be more than happy adding also cookies as authentication method
